test/Microsoft.Extensions.Logging.Testing.Tests/LoggedTestXunitRetryTests.cs

```diff
@@ -0,0 +1,101 @@
+
+using System;
+using System.Linq;
+using Microsoft.AspNetCore.Testing.xunit;
+using Xunit;
+
+namespace Microsoft.Extensions.Logging.Testing.Tests
+{
+    [RetryTest(nameof(RetryAllPredicate), "sample reason")]
+    public class LoggedTestXunitRetryTests : LoggedTest
+    {
+        [Fact]
+        public void CompletesWithoutRetryOnSuccess()
+        {
+            Assert.Equal(2, RetryContext.Limit);
+
+            // This assert would fail on the second run
+            Assert.Equal(0, RetryContext.CurrentIteration);
+        }
+
+        [Fact]
+        public void RetriesUntilSuccess()
+        {
+            // This assert will fail the first time but pass on the second
+            Assert.Equal(1, RetryContext.CurrentIteration);
+
+            // This assert will ensure a message is logged for retried tests.
+            Assert.Equal(1, TestSink.Writes.Count);
+            var loggedMessage = TestSink.Writes.ToArray()[0];
+            Assert.Equal(LogLevel.Warning, loggedMessage.LogLevel);
+            Assert.Equal($"{nameof(RetriesUntilSuccess)} failed and retry conditions are met, re-executing. The reason for failure is sample reason.", loggedMessage.Message);
+        }
+
+        [ConditionalFact]
+        [OSSkipCondition(OperatingSystems.Windows)]
+        [RetryTest(OperatingSystems.Windows, nameof(RetryAllPredicate), "sample reason", 3)]
+        public void RetryCountNotOverridenWhenOSDoesNotMatch()
+        {
+            Assert.Equal(2, RetryContext.Limit);
+        }
+
+        [ConditionalFact]
+        [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX)]
+        [RetryTest(OperatingSystems.Windows, nameof(RetryAllPredicate), "sample reason", 3)]
+        public void RetryCountOverridenWhenOSMatches()
+        {
+            Assert.Equal(3, RetryContext.Limit);
+        }
+
+        [ConditionalFact]
+        [RetryTest(nameof(RetryInvalidOperationExceptionPredicate), "sample reason")]
+        public void RetryIfPredicateIsTrue()
+        {
+            if (RetryContext.CurrentIteration == 0)
+            {
+                Logger.LogWarning("Throw on first iteration");
+                throw new Exception();
+            }
+
+            // This assert will ensure a message is logged for retried tests.
+            Assert.Equal(1, TestSink.Writes.Count);
+            var loggedMessage = TestSink.Writes.ToArray()[0];
+            Assert.Equal(LogLevel.Warning, loggedMessage.LogLevel);
+            Assert.Equal($"{nameof(RetryIfPredicateIsTrue)} failed and retry conditions are met, re-executing. The reason for failure is sample reason.", loggedMessage.Message);
+        }
+
+        // Static predicates are valid
+        private static bool RetryAllPredicate(Exception e)
+            => true;
+
+        // Instance predicates are valid
+        private bool RetryInvalidOperationExceptionPredicate(Exception e)
+            => TestSink.Writes.Any(m => m.Message.Contains("Throw on first iteration"));
```

Comment:
Never mind about the logged test parameter. I didn't see that you already supported instance methods as predicates. This is exactly what I want.